packages/components/tests/integration/components/hds/form/checkbox/group-test.js

```diff
@@ -66,6 +66,36 @@ module('Integration | Component | hds/form/checkbox/group', function (hooks) {
     assert.dom('.hds-form-group__error').doesNotExist();
   });
 
+  // REQUIRED AND OPTIONAL
+
+  test('it should append an indicator to the legend text and set the required attribute when user input is required', async function (assert) {
+    assert.expect(3);
+    await render(
+      hbs`<Hds::Form::Checkbox::Group @isRequired={{true}} as |G|>
+            <G.Legend>This is the legend</G.Legend>
+            <G.Checkbox::Field checked="checked" @value="abc123" as |F|>
+              <F.Label>This is the control label</F.Label>
+            </G.Checkbox::Field>
+          </Hds::Form::Checkbox::Group>`
+    );
+    assert.dom('legend .hds-form-required-indicator').exists();
+    assert.dom('legend .hds-form-required-indicator').hasText('Required');
+    assert.dom('input[required]').exists();
```

Comment:
maybe I would use something more explicit about what we're testing
```suggestion
    assert.dom('input').hasAttribute('required');
```